pxr/imaging/hdSt/materialNetwork.h

```diff
@@ -20,9 +20,57 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+class SdrRegistry;
 class HdStResourceRegistry;
 using HioGlslfxSharedPtr = std::shared_ptr<class HioGlslfx>;
 using HdSt_MaterialParamVector = std::vector<class HdSt_MaterialParam>;
+struct HdSt_MaterialFilterTask;
+using HdSt_MaterialFilterTaskSharedPtr =
+    std::shared_ptr<HdSt_MaterialFilterTask>;
+
+extern HdMaterialNode2 const*
+HdSt_GetTerminalNode(
+    HdMaterialNetwork2 const& network,
+    TfToken const& terminalName,
+    SdfPath * terminalNodePath);
+
+/// Encapsulates the input data for MaterialX codegen as well as metadata
+/// necessary for completing `HdStMaterial::Sync` based on the result of the
+/// codegen.
+/// This object can either live on the stack, if MaterialX codegen happens
+/// synchronously, or on the heap, owned by the respective Sprim, if the
+/// codegen happens in parallel tasks.
+///
+struct ARCH_EXPORT_TYPE HdSt_MaterialFilterTask final
```

Comment:
This feels like an implementation detail of the MaterialX code, and I don't think this belongs in hdSt/materialNetwork.h.  Also curious why we need ARCH_EXPORT_TYPE here.